### ISSUE TITLE
awk: can't open file STAT_RESULTS

### ISSUE THREAD

**Issue by creasty** [supercrabtree/k#49]
I got an error, and directories weren't displayed.

``` sh
in ~github/creasty/dotfiles on master *+
❯ k
awk: can't open file STAT_RESULTS
 source line number 1
total 32
-rwxr-xr-x  1 ykiwng staff 4389  8  7    19:18 | Brewfile
-rw-r--r--  1 ykiwng staff  685 30  5    19:45 | README.md
drwxr-xr-x  9 ykiwng staff  306 16 11     2014 |
drwxr-xr-x  5 ykiwng staff  170  2  6    21:51 |
drwxr-xr-x  5 ykiwng staff  170 31  5    22:41 |
drwxr-xr-x 34 ykiwng staff 1156  8  7    19:25 |
-rwxr-xr-x  1 ykiwng staff 2228  3  6    23:37 | bootstrap
drwxr-xr-x  9 ykiwng staff  306  7  6    21:40 |
drwxr-xr-x  9 ykiwng staff  306  7  6    21:40 |
drwxr-xr-x  8 ykiwng staff  272  9  5    14:32 |
drwxr-xr-x  5 ykiwng staff  170 30  5    19:45 |
drwxr-xr-x  3 ykiwng staff  102 31  5    13:04 |
drwxr-xr-x  4 ykiwng staff  136 16  6    10:15 |
drwxr-xr-x  3 ykiwng staff  102 11  7    19:35 |
drwxr-xr-x  3 ykiwng staff  102 30  6    22:57 |
drwxr-xr-x  6 ykiwng staff  204  8  7    21:08 |
drwxr-xr-x  3 ykiwng staff  102 31  5    12:43 |
drwxr-xr-x  3 ykiwng staff  102 30  5    19:45 |
drwxr-xr-x  4 ykiwng staff  136 30  5    19:45 |
drwxr-xr-x  3 ykiwng staff  102 31  5    12:53 |
drwxr-xr-x  3 ykiwng staff  102 30  5    19:45 |
drwxr-xr-x  9 ykiwng staff  306 31  5    13:06 |
drwxr-xr-x  6 ykiwng staff  204 30  5    19:45 |
drwxr-xr-x  3 ykiwng staff  102  5  7    19:04 |
drwxr-xr-x  6 ykiwng staff  204  8  7    19:18 |
drwxr-xr-x 19 ykiwng staff  646 11  7    20:04 |
drwxr-xr-x 14 ykiwng staff  476 14  7    13:03 +
```

Here's my environment:

``` sh
in ~github/creasty/dotfiles on master *+
❯ zsh --version
zsh 5.0.8 (x86_64-apple-darwin14.3.0)

in ~github/creasty/dotfiles on master *+
❯ git --version
git version 2.4.5

in ~github/creasty/dotfiles on master *+
❯ ls -la
total 80
drwxr-xr-x  35 ykiwng  staff  1190  7 14 13:02 .
drwxr-xr-x  44 ykiwng  staff  1496  7 14 12:16 ..
-rw-r--r--@  1 ykiwng  staff  8196  7  5 19:04 .DS_Store
drwxr-xr-x  20 ykiwng  staff   680  7 14 13:05 .git
-rw-r--r--   1 ykiwng  staff   186  5 30 19:45 .gitignore
-rw-r--r--   1 ykiwng  staff   721  7 14 13:02 .gitmodules
-rw-r--r--   1 ykiwng  staff     6  2  7 22:40 .ruby-version
drwxr-xr-x   3 ykiwng  staff   102  4 25 21:32 .vagrant
-rwxr-xr-x   1 ykiwng  staff  4389  7  8 19:18 Brewfile
-rw-r--r--   1 ykiwng  staff   685  5 30 19:45 README.md
drwxr-xr-x   9 ykiwng  staff   306 11 16  2014 app
drwxr-xr-x   5 ykiwng  staff   170  6  2 21:51 aws
drwxr-xr-x   5 ykiwng  staff   170  5 31 22:41 bash
drwxr-xr-x  34 ykiwng  staff  1156  7  8 19:25 bin
-rwxr-xr-x   1 ykiwng  staff  2228  6  3 23:37 bootstrap
drwxr-xr-x   9 ykiwng  staff   306  6  7 21:40 bootstrap.d
drwxr-xr-x   9 ykiwng  staff   306  6  7 21:40 config
drwxr-xr-x   8 ykiwng  staff   272  5  9 14:32 docs
drwxr-xr-x   5 ykiwng  staff   170  5 30 19:45 dotssh
drwxr-xr-x   3 ykiwng  staff   102  5 31 13:04 elasticsearch
drwxr-xr-x   4 ykiwng  staff   136  6 16 10:15 git
drwxr-xr-x   3 ykiwng  staff   102  7 11 19:35 golang
drwxr-xr-x   3 ykiwng  staff   102  6 30 22:57 haskell
drwxr-xr-x   6 ykiwng  staff   204  7  8 21:08 lib
drwxr-xr-x   3 ykiwng  staff   102  5 31 12:43 lua
drwxr-xr-x   3 ykiwng  staff   102  5 30 19:45 nodejs
drwxr-xr-x   4 ykiwng  staff   136  5 30 19:45 peco
drwxr-xr-x   3 ykiwng  staff   102  5 31 12:53 python
drwxr-xr-x   3 ykiwng  staff   102  5 30 19:45 r
drwxr-xr-x   9 ykiwng  staff   306  5 31 13:06 ruby
drwxr-xr-x   6 ykiwng  staff   204  5 30 19:45 tags
drwxr-xr-x   3 ykiwng  staff   102  7  5 19:04 tex
drwxr-xr-x   6 ykiwng  staff   204  7  8 19:18 tmux
drwxr-xr-x  19 ykiwng  staff   646  7 11 20:04 vim
drwxr-xr-x  14 ykiwng  staff   476  7 14 13:03 zsh
```

**Comment by subshad** [comment]
I just ran into this. In your setup, are you using a global alias like so:

```zsh
alias -g A="| awk"
```

It looks like the [typeset](https://github.com/supercrabtree/k/blob/master/k.sh#L152) is creating a variable A, but since zsh uses a global alias for " A" to " | awk" this typeset is actually being interpreted as

```zsh
typeset -a MAX_LEN | awk RESULTS STAT_RESULTS
```

resulting in awk trying to access the file STAT_RESULTS which obviously doesn't exist. Not sure how to fix this short of changing that global alias...

Sorry to necro-post, thought it would be useful since I just ran into this, and I'm sure that that's a common alias.
